test/object-store/util/sync/baas_admin_api.cpp

```diff
@@ -341,6 +389,203 @@ app::Response do_http_request(const app::Request& request)
     };
 }
 
+class Baasaas {
+public:
+    enum class StartMode { Default, GitHash, Branch, PatchId };
+    explicit Baasaas(std::string api_key, StartMode mode, std::string ref_spec)
+        : m_api_key(std::move(api_key))
+    {
+        auto logger = util::Logger::get_default_logger();
+        std::string url_path = "startContainer";
+        if (mode == StartMode::GitHash) {
+            url_path = util::format("startContainer?githash=%1", ref_spec);
+            logger->info("Starting baasaas container with githash of %1", ref_spec);
+        }
+        else if (mode == StartMode::Branch) {
+            url_path = util::format("startContainer?branch=%1", ref_spec);
+            logger->info("Starting baasaas container on branch %1", ref_spec);
+        }
+        else if (mode == StartMode::PatchId) {
+            url_path = util::format("startContainer?patchId=%1", ref_spec);
+            logger->info("Starting baasaas container for patch id %1", ref_spec);
+        }
+        else {
+            logger->info("Starting baasaas container");
+        }
+
+        auto resp = do_request(std::move(url_path), app::HttpMethod::post);
+        m_container_id = resp["id"].get<std::string>();
+        logger->info("Baasaas container started with id \"%1\"", m_container_id);
+    }
+    Baasaas(const Baasaas&) = delete;
+    Baasaas(Baasaas&&) = delete;
+    Baasaas& operator=(const Baasaas&) = delete;
+    Baasaas& operator=(Baasaas&&) = delete;
+
+    ~Baasaas()
+    {
+        stop();
```

Comment:
Does baasaas have a TTL or something just in case our tests on PRs crash and leave a bunch of containers running?

Reply:
Yes, they'll get reaped automatically after about 60 minutes.
Ideally, it would be great to guarantee these get cleaned up even if the test process itself dies. Maybe we could have the setup write the container ID to a file that the evergreen task could check for in the "post" task, or something like that. Seems like this should be relatively uncommon though so I think it would be fine to just revisit this later.

Reply:
I've added a lock file to this type so it will create a file called `baasaas_instance.lock` in the working directory of the tests and remove it after the instance is stopped. I also added the ability to specify a container ID so the tests can still interact with baasaas to get their base urls and stuff, but won't tear down the container if it was started manually to make this easier to work with the baasaas cli script.